huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go

```diff
@@ -0,0 +1,219 @@
+	}
+
+	instanceID, replicationID, err := extractAttachId(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	n, err := protectedinstances.Get(client, instanceID).Extract()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			if resp, pErr := common.ParseErrorMsg(errCode.Body); pErr == nil && resp.ErrorCode == "SDRS.0208" {
+				// `SDRS.0208` means invalid protected instance ID
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{},
+					"error retrieving SDRS protected instance when query replication attach")
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	attachment, err := flattenReplicationAttach(n, replicationID)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving SDRS replication attach")
+	}
+
+	mErr := multierror.Append(
+		nil,
+		d.Set("region", region),
+		d.Set("device", attachment.Device),
+		d.Set("replication_id", attachment.Replication),
+		d.Set("status", n.Status),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenReplicationAttach(instance *protectedinstances.Instance,
+	replicationID string) (*protectedinstances.Attachment, error) {
+	for _, attach := range instance.Attachment {
+		if attach.Replication == replicationID {
+			// find the target attachment
+			return &attach, nil
+		}
+	}
+	return nil, golangsdk.ErrDefault404{}
+}
+
+func resourceReplicationAttachDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+	// use for WaitForJobSuccess
+	client.Endpoint = client.ResourceBase
+
+	instanceID, replicationID, err := extractAttachId(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	n, err := attachreplication.Delete(client, instanceID, replicationID).ExtractJobResponse()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			resp, pErr := common.ParseErrorMsg(errCode.Body)
+			if pErr != nil {
+				log.Printf("[ERROR] error parse SDRS replication attach delete error, %s", pErr)
+				return diag.FromErr(err)
+			}
+
+			if resp.ErrorCode == "SDRS.0208" || resp.ErrorCode == "SDRS.0213" {
+				// `SDRS.0208` means invalid protected instance ID
+				// `SDRS.0213` means invalid replication pair ID
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{},
+					"error deleting SDRS replication attach")
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	deleteTimeoutSec := int(d.Timeout(schema.TimeoutDelete).Seconds())
+	if err := attachreplication.WaitForJobSuccess(client, deleteTimeoutSec, n.JobID); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func formatAttachId(instanceID string, replicationID string) string {
+	return fmt.Sprintf("%s:%s", instanceID, replicationID)
+}
+
+func extractAttachId(resourceID string) (instanceID string, replicationID string, err error) {
```

Comment:
It is not recommended to specify variables when defining the return type.

Reply:
Defining variable names is to distinguish multiple return values of the same type.